diff --git a/components/pthread/CMakeLists.txt b/components/pthread/CMakeLists.txt
index 653799ad..df5a961a 100644
--- a/components/pthread/CMakeLists.txt
+++ b/components/pthread/CMakeLists.txt
@@ -4,7 +4,7 @@ idf_component_register(SRCS "pthread.c"
                     INCLUDE_DIRS include)
 
 set(extra_link_flags "-u pthread_include_pthread_impl")
-list(APPEND extra_link_flags "-u pthread_include_pthread_cond_impl")
+list(APPEND extra_link_flags "-u pthread_include_pthread_cond_var_impl")
 list(APPEND extra_link_flags "-u pthread_include_pthread_local_storage_impl")
 
 if(CONFIG_FREERTOS_ENABLE_STATIC_TASK_CLEAN_UP)
